server/src/main/java/org/candlepin/dto/DTOFactory.java

```diff
@@ -0,0 +1,259 @@
+     *  registered
+     */
+    public DTOTranslator unregisterTranslator(Class srcClass) {
+        if (srcClass == null) {
+            throw new IllegalArgumentException("srcClass is null");
+        }
+
+        return this.translators.remove(srcClass);
+    }
+
+    /**
+     * Retrieves the translator registered for the specified class. If a translator has not been
+     * registered for the given class, this method returns null.
+     * <p></p>
+     * Note that unlike the fetchTranslator methods, this method will only perform an exact class
+     * match, and will not attempt to find a suitable translator if the given class does not have
+     * an explicitly registered translator.
+     *
+     * @param srcClass
+     *  The model class for which to retrieve a translator
+     *
+     * @throw IllegalArgumentException
+     *  if srcClass is null
+     *
+     * @return
+     *  The translator registered for the given class, or null if a translator has not yet been
+     *  registered
+     */
+    public DTOTranslator getTranslator(Class srcClass) {
+        if (srcClass == null) {
+            throw new IllegalArgumentException("srcClass is null");
+        }
+
+        return this.translators.get(srcClass);
+    }
+
+    /**
+     * Fetches a translator for the given class. If a translator cannot be found, this method
+     * returns null.
+     * <p></p>
+     * This method uses the following algorithm to determine which translator to use:
+     * <pre>
+     * 1. Fetch the source object's class, C
+     * 2. Fetch any translator T registered for class C
+     * 3. If T is not null, go to step X
+     * 4. Fetch the list of interfaces, L, implemented by class C
+     * 5. For each interface I in list L...
+     *    a. fetch the translator T registered for interface I
+     *    b. if T is not null, go to step X
+     * 6. Set C to the superclass of C
+     * 7. If C is not null, go to step 2
+     * 8. Return T
+     * </pre>
+     *
+     * @param srcClass
+     *  The source class for which to find a translator
+     *
+     * @throws IllegalArgumentException
+     *  if srcClass is null
+     *
+     * @return
+     *  a translator for the given source object, or null if a translator could not be found
+     */
+    public DTOTranslator fetchTranslatorByClass(Class srcClass) {
+        if (srcClass == null) {
+            throw new IllegalArgumentException("srcClass is null");
+        }
+
+        DTOTranslator translator = null;
+
+        for (; srcClass != null && translator == null; srcClass = srcClass.getSuperclass()) {
+            translator = this.translators.get(srcClass);
+
+            // Check interfaces, if necessary
+            if (translator == null) {
+                translator = this.fetchTranslatorByInterfaces(srcClass);
+            }
+        }
+
+        return translator;
+    }
+
+    /**
+     * Recursive implementation for fetching translators by interface for a given class.
```

Comment:
Kind of. It's only even recursive due to how identifying interfaces works. It still checks the class hierarchy first, then the interfaces in declaration (and then hierarchy) order. But because getParentClass doesn't work with interfaces, it has to be implemented the way it is to walk up the hierarchy of each implemented given interface.